homeassistant/components/lidarr/sensor.py

```diff
@@ -0,0 +1,161 @@
+
+from . import LidarrEntity
+from .const import BYTE_SIZES, DEFAULT_NAME, DOMAIN
+from .coordinator import LidarrDataUpdateCoordinator
+
+SENSOR_TYPES: tuple[SensorEntityDescription, ...] = (
+    SensorEntityDescription(
+        key="commands",
+        name="Commands",
+        native_unit_of_measurement="Commands",
+        icon="mdi:code-braces",
+    ),
+    SensorEntityDescription(
+        key="diskspace",
+        name="Disk Space",
+        native_unit_of_measurement=DATA_GIGABYTES,
+        icon="mdi:harddisk",
+    ),
+    SensorEntityDescription(
+        key="queue",
+        name="Queue",
+        native_unit_of_measurement="Albums",
+        icon="mdi:music",
+    ),
+    SensorEntityDescription(
+        key="status",
+        name="Status",
+        native_unit_of_measurement="Status",
+        icon="mdi:information",
+    ),
+    SensorEntityDescription(
+        key="wanted",
+        name="Wanted",
+        native_unit_of_measurement="Albums",
+        icon="mdi:music",
+    ),
+    SensorEntityDescription(
+        key="upcoming",
+        name="Upcoming",
+        native_unit_of_measurement="Albums",
+        icon="mdi:music",
+    ),
+)
+
+PARALLEL_UPDATES = 1
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Lidarr sensors based on a config entry."""
+    async_add_entities(
+        LidarrSensor(hass.data[DOMAIN][entry.entry_id], description)
+        for description in SENSOR_TYPES
+    )
+
+
+class LidarrSensor(LidarrEntity, SensorEntity):
+    """Implementation of the Lidarr sensor."""
+
+    coordinator: LidarrDataUpdateCoordinator
+
+    def __init__(
+        self,
+        coordinator: LidarrDataUpdateCoordinator,
+        description: SensorEntityDescription,
+    ) -> None:
+        """Create Lidarr entity."""
+        super().__init__(coordinator)
+        self.entity_description = description
+        self._attr_name = f"{DEFAULT_NAME} {description.name}"
+        self._attr_unique_id = f"{coordinator.config_entry.entry_id}_{description.key}"
+
+    @property
+    def extra_state_attributes(self) -> dict[str, StateType | datetime]:
+        """Return the state attributes of the sensor."""
+        if self.entity_description.key == "commands":
+            return {c.name: c.status for c in self.coordinator.commands}
+        if self.entity_description.key == "diskspace":
+            return {m.path: mnt_str(m) for m in self.coordinator.disk_space}
+        if self.entity_description.key == "queue":
+            return {i.title: queue_str(i) for i in self.coordinator.queue.records}
+        if self.entity_description.key == "status":
+            return self.coordinator.system_status.attributes
+        if self.entity_description.key == "upcoming":
+            return to_attr(self.coordinator.calendar)
+        return to_attr(self.coordinator.wanted.records)
```

Comment:
Similar comment to Radarr, this should be dropped.

A state with many attributes should instead be turned into individual sensor entities. In the case of the calendar it should be a calendar platform.